### PR TITLE
Move KV integrity checker out of the sdk [kvl-1418]

### PR DESCRIPTION
**Pull request by nicu-da**
changelog_begin
changelog_end

### CODE REVIEW

**Comment by garyverhaegen-da** [comment]
Is this getting deleted or has it moved elsewhere?

I'm a bit confused by the CHANGELOG entry: is this a user-visible tool? If so, we can't just remove it. It's fine to move the code away but the artifacts still need to be available somewhere somehow.

**Comment by nicu-da** [comment]
> 

@garyverhaegen-da I'm not sure how much visibility this tool had to any users, as it was KV specific. 
The artifacts for the old version would remain though, it's getting moved to the internal KV repo. 

@miklos-da Are we aware of any external uses of this tool? As far as I knew it was basically internal only.

**Review by garyverhaegen-da — APPROVED** [review]
Normally we only put CHANGELOG entries for user-affecting changes. If this was never part of our supported public interfaces it probably shouldn't be in the release notes. I'd leave it in the CHANGELOG though, and let product take it out when they compile the release notes.

It would also be nice to have a link in this PR to the new place where this code lives, for future tracking.

**Comment by nicu-da** [comment]
> Normally we only put CHANGELOG entries for user-affecting changes. If this was never part of our supported public interfaces it probably shouldn't be in the release notes. I'd leave it in the CHANGELOG though, and let product take it out when they compile the release notes.
> 
> It would also be nice to have a link in this PR to the new place where this code lives, for future tracking.

Good points @garyverhaegen-da . 
When merging I will remove the changelog so that it's cleaner.

I'm not sure how do we link from public repos to private repos though. I've already added the link to this PR in the private repo PR, should we do the reverse as well?

**Review by fabiotudone-da — APPROVED** [review]

**Comment by miklos-da** [comment]
> > Is this getting deleted or has it moved elsewhere?
> > I'm a bit confused by the CHANGELOG entry: is this a user-visible tool? If so, we can't just remove it. It's fine to move the code away but the artifacts still need to be available somewhere somehow.
> 
> @garyverhaegen-da I'm not sure how much visibility this tool had to any users, as it was KV specific. The artifacts for the old version would remain though, it's getting moved to the internal KV repo.
> 
> @miklos-da Are we aware of any external uses of this tool? As far as I knew it was basically internal only.

Yes, this is an internal tool and there should be no external uses of it. Let's remove the CHANGELOG entry.

**Review by hubert-da — APPROVED** [review]

File: ledger/participant-state/kvutils/tools/README.md [hunk 1]
```diff
@@ -1,37 +1,3 @@
-# `integrity-check`
```
Comment (miklos-da) [inline ledger/participant-state/kvutils/tools/README.md:1]: We might want to move this module to somewhere else later as this only contain the submission extractor (i.e., once we have moved the rest of the kvutils module out).

**Review by miklos-da — APPROVED** [review]

File: ledger/participant-state/kvutils/tools/README.md (hunk 1 above)
Comment (hubert-da) [reply ledger/participant-state/kvutils/tools/README.md:1]: I will take care of that.

**Comment by hubert-da** [comment]
Let's hold off merging this PR. I'd like to make all the PRs ready on the integration kit side first.

**Comment by hubert-da** [comment]
Superseded by https://github.com/digital-asset/daml/pull/13704

**Pull request closed without merging**